### PR TITLE
Add support for aria-describedby to Table component

### LINKED ISSUE #1143: Add support for <caption> tags and aria-describedby attribute to the Table component
Allow users to set a `<caption>` on the Table component, and add an `aria-describedby` attribute which could point to a summary of the table.

One, or the other, or both could be provided as additional props on the Table component.

### PR DESCRIPTION
**Pull request by ianoxley**
Add an `ariaDescribedBy` prop to the `Table` component.

This is a string, which points to the HTML id attribute of an element containing a description of the table.

When the `Table` component renders it renders an `aria-describedby` attribute using the value of the `ariaDescribedBy` prop.

Add ariaDescribedBy prop to the table definition.

Update release notes.

## Related Issues / Pull Requests
Resolves #1143 

## Testing Instructions
1. Create a `<Table>` component with the `ariaDescribedBy` prop set to the HTML id of the element on the page that contains a description for the table.

2. The rendered `<table>` element should contain an `aria-describedby` attribute with the id that was passed in as the `ariaDescribedBy` prop.

### CODE REVIEW

**Comment by ianoxley** [comment]
@andrewjtait Updated release notes to target 1.6

File: src/components/table/table.js [hunk 1]
```diff
@@ -274,6 +274,15 @@ class Table extends React.Component {
     caption: PropTypes.string,
 
     /**
+     * The HTML id of the element that contains a description
+     * of this table.
+     *
+     * @property ariaDescribedBy
+     * @type string
+     */
+    ariaDescribedBy: PropTypes.string,
```
Comment (andrewjtait) [inline src/components/table/table.js]: Do you think we should match the React API prop name `aria-describedby`?

Comment (SeanArmstrong) [reply src/components/table/table.js]: Yes i think its easier than converting all the tags

**Comment by andrewjtait** [comment]
QA'd and works as expected ✅

**Pull request merged**